ethereum/rpc/namespaces/eth/api.go

```diff
@@ -689,11 +684,24 @@ func (e *PublicAPI) GetBlockByNumber(ethBlockNum rpctypes.BlockNumber, fullTx bo
 	return e.backend.GetBlockByNumber(ethBlockNum, fullTx)
 }
 
+// GetTxByEthHash uses `/tx_query` to find transaction by ethereum tx hash
+func GetTxByEthHash(ctx context.Context, client rpcclient.Client, hash common.Hash) (*tmrpctypes.ResultTx, error) {
+	query := fmt.Sprintf("ethereum_tx.ethereumTxHash='%s'", hash.Hex())
```

Comment:
can we use constants instead? This could potentially break if we rename the event. Should be sth like the following:

`fmt.Sprintf("%s.%s='%s'", const1, eventkey, hash.Hex())`

Reply:
good idea, done